internal/nginx/nginx.go

```diff
@@ -187,12 +188,13 @@ func NewUpstreamWithDefaultServer(name string) Upstream {
 				Port:        "8181",
 				MaxFails:    1,
 				FailTimeout: "10s",
-			}},
+			},
+		},
 	}
 }
 
 // NewNginxController creates a NGINX controller
-func NewNginxController(nginxConfPath string, local bool) *Controller {
+func NewNginxController(nginxConfPath string, local bool, nginxBinary string) *Controller {
```

Comment:
please make `nginxBinary` the second argument. This way we'll have two path-related arguments as the first and the second and a boolean as the last one.